23.05.0: Orange Pi R1 Plus LTS: eth0 not working

### Describe the bug

No matter how configured eth0 is not passing (neither sending nor receiving) traffic. Link state is shown correctly though.

eth1 is working as it should.

I was previously using the Xulong fork (https://github.com/orangepi-xunlong/openwrt) where both interfaces work flawlessly.

### OpenWrt version

r23497-6637af95aa

### OpenWrt target/subtarget

rockchip/armv8

### Device

Xunlong Orange Pi R1 Plus LTS

### Image kind

Official downloaded image

### Steps to reproduce

Flash the official 23.05 squashfs image.

Configure eth0 as static LAN interface. For example with that network config (to have both devices in bridged mode as lan):
```
config interface 'loopback'
	option device 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fd14:eac3:1e2c::/48'

config device
	option name 'br-lan'
	option type 'bridge'
	list ports 'eth0'
	list ports 'eth1'

config interface 'lan'
	option device 'br-lan'
	option proto 'static'
	option ipaddr '192.168.1.1'
	option netmask '255.255.255.0'
	option ip6assign '60'
```
(I also tried various other setups like other subnet/IP for eth0, only configuring eth0, my working config from the xulong fork, ...)

All other configuration untouched.

Connect a computer to eth0. Set a static IP on the computer since DHCP will not work.

Try to ping in either direction.

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

Full serial log:

```
U-Boot TPL 2021.07-OpenWrt-r23497-6637af95aa (Oct 09 2023 - 21:45:35)
LPDDR3, 333MHz
BW=32 Col=10 Bk=8 CS0 Row=14 CS1 Row=14 CS=2 Die BW=16 Size=1024MB
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2021.07-OpenWrt-r23497-6637af95aa (Oct 09 2023 - 21:45:35 +0000)
Trying to boot from MMC1
NOTICE:  BL31: v2.3():v2.3
NOTICE:  BL31: Built : 15:56:43, Apr 20 2020
NOTICE:  BL31:Rockchip release version: v1.2


U-Boot 2021.07-OpenWrt-r23497-6637af95aa (Oct 09 2023 - 21:45:35 +0000)

Model: Xunlong Orange Pi R1 Plus LTS
DRAM:  1022 MiB
PMIC:  RK8050 (on=0x40, off=0x00)
MMC:   mmc@ff500000: 1
Loading Environment from MMC... MMC Device 0 not found
*** Warning - No MMC card found, using default environment

In:    serial@ff130000
Out:   serial@ff130000
Err:   serial@ff130000
Model: Xunlong Orange Pi R1 Plus LTS
Net:   eth0: ethernet@ff540000
Hit any key to stop autoboot:  0 
MMC Device 0 not found
no mmc device at slot 0
switch to partitions #0, OK
mmc1 is current device
Scanning mmc 1:1...
Found U-Boot script /boot.scr
534 bytes read in 3 ms (173.8 KiB/s)
## Executing script at 00500000
4131744 bytes read in 182 ms (21.6 MiB/s)
## Loading kernel from FIT Image at 02080000 ...
   Using 'config-1' configuration
   Trying 'kernel-1' kernel subimage
     Description:  ARM64 OpenWrt Linux-5.15.134
     Type:         Kernel Image
     Compression:  lzma compressed
     Data Start:   0x020800ec
     Data Size:    4093951 Bytes = 3.9 MiB
     Architecture: AArch64
     OS:           Linux
     Load Address: 0x03200000
     Entry Point:  0x03200000
     Hash algo:    crc32
     Hash value:   f6ed92b7
     Hash algo:    sha1
     Hash value:   b1cb8d9637cd7eae4e0b00dac0d1d12b7a1607ae
   Verifying Hash Integrity ... crc32+ sha1+ OK
## Loading fdt from FIT Image at 02080000 ...
   Using 'config-1' configuration
   Trying 'fdt-1' fdt subimage
     Description:  ARM64 OpenWrt xunlong_orangepi-r1-plus-lts device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x02467a38
     Data Size:    35859 Bytes = 35 KiB
     Architecture: AArch64
     Hash algo:    crc32
     Hash value:   9b12ec37
     Hash algo:    sha1
     Hash value:   49f438614ae6eceb68d9d6bd70c3497d76526bf6
   Verifying Hash Integrity ... crc32+ sha1+ OK
   Booting using the fdt blob at 0x2467a38
   Uncompressing Kernel Image
   Loading Device Tree to 000000003df25000, end 000000003df30c12 ... OK

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.15.134 (builder@buildhost) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r23497-6637af95aa) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP PREEMPT Mon Oct 9 21:45:35 2023
[    0.000000] Machine model: Xunlong Orange Pi R1 Plus LTS
[    0.000000] earlycon: uart8250 at MMIO32 0x00000000ff130000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] On node 0, zone DMA: 512 pages in unavailable ranges
[    0.000000] cma: Reserved 8 MiB at 0x000000003e400000
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 18 pages/cpu s32792 r8192 d32744 u73728
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 257544
[    0.000000] Kernel command line: console=ttyS2,1500000 console=tty1 earlycon=uart8250,mmio32,0xff130000 swiotlb=1 root=PARTUUID=5452574f-02 rw rootwait
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 1002136K/1046528K available (8960K kernel code, 2392K rwdata, 1736K rodata, 1664K init, 737K bss, 36200K reserved, 8192K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu: 	RCU event tracing is enabled.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.000000] 	Trampoline variant of Tasks RCU enabled.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: 0xffff8000084139d0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000001] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.001360] Console: colour dummy device 80x25
[    0.001791] printk: console [tty1] enabled
[    0.002188] printk: bootconsole [uart8250] disabled
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.15.134 (builder@buildhost) (aarch64-openwrt-linux-musl-gcc (OpenWrt GCC 12.3.0 r23497-6637af95aa) 12.3.0, GNU ld (GNU Binutils) 2.40.0) #0 SMP PREEMPT Mon Oct 9 21:45:35 2023
[    0.000000] Machine model: Xunlong Orange Pi R1 Plus LTS
[    0.000000] earlycon: uart8250 at MMIO32 0x00000000ff130000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x000000003fffffff]
[    0.000000] On node 0, zone DMA: 512 pages in unavailable ranges
[    0.000000] cma: Reserved 8 MiB at 0x000000003e400000
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 18 pages/cpu s32792 r8192 d32744 u73728
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 257544
[    0.000000] Kernel command line: console=ttyS2,1500000 console=tty1 earlycon=uart8250,mmio32,0xff130000 swiotlb=1 root=PARTUUID=5452574f-02 rw rootwait
[    0.000000] Dentry cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 1002136K/1046528K available (8960K kernel code, 2392K rwdata, 1736K rodata, 1664K init, 737K bss, 36200K reserved, 8192K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu: 	RCU event tracing is enabled.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.000000] 	Trampoline variant of Tasks RCU enabled.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: 0xffff8000084139d0
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000001] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.001360] Console: colour dummy device 80x25
[    0.001791] printk: console [tty1] enabled
[    0.002188] printk: bootconsole [uart8250] disabled
[    0.002718] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=240000)
[    0.002757] pid_max: default: 32768 minimum: 301
[    0.003471] Mount-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.003512] Mountpoint-cache hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.006591] rcu: Hierarchical SRCU implementation.
[    0.007806] smp: Bringing up secondary CPUs ...
[    0.008705] Detected VIPT I-cache on CPU1
[    0.008811] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.009663] Detected VIPT I-cache on CPU2
[    0.009744] CPU2: Booted secondary processor 0x0000000002 [0x410fd034]
[    0.010615] Detected VIPT I-cache on CPU3
[    0.010698] CPU3: Booted secondary processor 0x0000000003 [0x410fd034]
[    0.010832] smp: Brought up 1 node, 4 CPUs
[    0.010938] SMP: Total of 4 processors activated.
[    0.010957] CPU features: detected: 32-bit EL0 Support
[    0.010977] CPU features: detected: CRC32 instructions
[    0.011071] CPU features: emulated: Privileged Access Never (PAN) using TTBR0_EL1 switching
[    0.013377] CPU: All CPU(s) started at EL2
[    0.013417] alternatives: patching kernel code
[    0.026488] KASLR disabled due to lack of seed
[    0.026727] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.026780] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.027521] pinctrl core: initialized pinctrl subsystem
[    0.029716] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.032242] DMA: preallocated 128 KiB GFP_KERNEL pool for atomic allocations
[    0.032541] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.032783] DMA: preallocated 128 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.033996] thermal_sys: Registered thermal governor 'step_wise'
[    0.034012] thermal_sys: Registered thermal governor 'power_allocator'
[    0.034453] cpuidle: using governor menu
[    0.034616] ASID allocator initialised with 65536 entries
[    0.034803] Serial: AMBA PL011 UART driver
[    0.052679] rockchip-gpio ff210000.gpio0: probed /pinctrl/gpio0@ff210000
[    0.053609] rockchip-gpio ff220000.gpio1: probed /pinctrl/gpio1@ff220000
[    0.054238] rockchip-gpio ff230000.gpio2: probed /pinctrl/gpio2@ff230000
[    0.054873] rockchip-gpio ff240000.gpio3: probed /pinctrl/gpio3@ff240000
[    0.065923] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.065982] HugeTLB registered 32.0 MiB page size, pre-allocated 0 pages
[    0.066007] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.066030] HugeTLB registered 64.0 KiB page size, pre-allocated 0 pages
[    0.067774] cryptd: max_cpu_qlen set to 1000
[    0.071116] vdd_5v_lan: supplied by vcc_sys
[    0.072095] iommu: Default domain type: Passthrough 
[    0.073374] SCSI subsystem initialized
[    0.073735] usbcore: registered new interface driver usbfs
[    0.073826] usbcore: registered new interface driver hub
[    0.073896] usbcore: registered new device driver usb
[    0.074076] pps_core: LinuxPPS API ver. 1 registered
[    0.074098] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.074142] PTP clock support registered
[    0.076777] clocksource: Switched to clocksource arch_sys_counter
[    0.077240] VFS: Disk quotas dquot_6.6.0
[    0.077358] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.083589] NET: Registered PF_INET protocol family
[    0.083873] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.085686] tcp_listen_portaddr_hash hash table entries: 512 (order: 1, 8192 bytes, linear)
[    0.085780] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.085820] TCP established hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.085954] TCP bind hash table entries: 8192 (order: 5, 131072 bytes, linear)
[    0.086170] TCP: Hash tables configured (established 8192 bind 8192)
[    0.086405] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.086481] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.086972] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.087046] PCI: CLS 0 bytes, default 64
[    0.093283] workingset: timestamp_bits=46 max_order=18 bucket_order=0
[    0.104363] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.104425] jffs2: version 2.2 (NAND) (SUMMARY) (ZLIB) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.106865] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 248)
[    0.107186] io scheduler mq-deadline registered
[    0.110902] gpio-syscon ff100000.syscon:gpio: can't read the data register offset!
[    0.115033] dma-pl330 ff1f0000.dma-controller: Loaded driver for PL330 DMAC-241330
[    0.115099] dma-pl330 ff1f0000.dma-controller: 	DBUFF-128x8bytes Num_Chans-8 Num_Peri-20 Num_Events-16
[    0.116056] Serial: 8250/16550 driver, 16 ports, IRQ sharing enabled
[    0.120390] printk: console [ttyS2] disabled
[    0.120522] ff130000.serial: ttyS2 at MMIO 0xff130000 (irq = 18, base_baud = 1500000) is a 16550A
[    0.185423] printk: console [ttyS2] enabled
[    0.193784] loop: module loaded
[    0.194119] mtip32xx Version 1.3.1
[    0.194664] Loading iSCSI transport class v2.0-870.
[    0.199053] spi-nor spi0.0: unrecognized JEDEC id bytes: 5e 40 18 5e 40 18
[    0.199720] spi-nor: probe of spi0.0 failed with error -2
[    0.203225] rk_gmac-dwmac ff540000.ethernet: IRQ eth_wake_irq not found
[    0.203861] rk_gmac-dwmac ff540000.ethernet: IRQ eth_lpi not found
[    0.204579] rk_gmac-dwmac ff540000.ethernet: PTP uses main clock
[    0.206651] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    0.207272] ehci-pci: EHCI PCI platform driver
[    0.207739] ehci-platform: EHCI generic platform driver
[    0.210718] ehci-platform ff5c0000.usb: EHCI Host Controller
[    0.211271] ehci-platform ff5c0000.usb: new USB bus registered, assigned bus number 1
[    0.212127] ehci-platform ff5c0000.usb: irq 37, io mem 0xff5c0000
[    0.236613] ehci-platform ff5c0000.usb: USB 2.0 started, EHCI 1.00
[    0.237850] hub 1-0:1.0: USB hub found
[    0.238237] hub 1-0:1.0: 1 port detected
[    0.239324] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    0.239927] ohci-platform: OHCI generic platform driver
[    0.240896] ohci-platform ff5d0000.usb: Generic Platform OHCI controller
[    0.241528] ohci-platform ff5d0000.usb: new USB bus registered, assigned bus number 2
[    0.242407] ohci-platform ff5d0000.usb: irq 38, io mem 0xff5d0000
[    0.311288] hub 2-0:1.0: USB hub found
[    0.311711] hub 2-0:1.0: 1 port detected
[    0.313381] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    0.313922] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 3
[    0.314836] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000000002010010
[    0.315756] xhci-hcd xhci-hcd.0.auto: irq 39, io mem 0xff600000
[    0.316558] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    0.317155] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 4
[    0.317859] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    0.319175] hub 3-0:1.0: USB hub found
[    0.319568] hub 3-0:1.0: 1 port detected
[    0.320418] usb usb4: We don't know the algorithms for LPM for this host, disabling LPM.
[    0.321706] hub 4-0:1.0: USB hub found
[    0.322086] hub 4-0:1.0: 1 port detected
[    0.323036] usbcore: registered new interface driver usb-storage
[    0.323817] i2c_dev: i2c /dev entries driver
[    0.327217] rk808 1-0018: chip id: 0x8050
[    0.332957] rk808-regulator rk808-regulator: there is no dvs0 gpio
[    0.333579] rk808-regulator rk808-regulator: there is no dvs1 gpio
[    0.336003] vdd_log: supplied by vcc_sys
[    0.338578] vdd_arm: supplied by vcc_sys
[    0.339266] vcc_ddr: supplied by vcc_sys
[    0.340843] vcc_io: supplied by vcc_sys
[    0.343240] vcc_18: supplied by vcc_io
[    0.347470] vcc18_emmc: supplied by vcc_io
[    0.351790] vdd_10: supplied by vcc_sys
[    0.363522] rk808-rtc rk808-rtc: registered as rtc0
[    0.365712] rk808-rtc rk808-rtc: setting system clock to 2023-10-09T21:49:58 UTC (1696888198)
[    0.370026] dw_wdt ff1a0000.watchdog: No valid TOPs array specified
[    0.372767] cpu cpu0: EM: created perf domain
[    0.375192] sdhci: Secure Digital Host Controller Interface driver
[    0.375784] sdhci: Copyright(c) Pierre Ossman
[    0.376179] Synopsys Designware Multimedia Card Interface Driver
[    0.377051] sdhci-pltfm: SDHCI platform and OF driver helper
[    0.377566] dwmmc_rockchip ff500000.mmc: IDMAC supports 32-bit address mode.
[    0.377603] dwmmc_rockchip ff500000.mmc: Using internal DMA controller.
[    0.378811] dwmmc_rockchip ff500000.mmc: Version ID is 270a
[    0.379381] dwmmc_rockchip ff500000.mmc: DW MMC controller at irq 34,32 bit host data width,256 deep fifo
[    0.380583] ledtrig-cpu: registered to indicate activity on CPUs
[    0.381395] hid: raw HID events driver (C) Jiri Kosina
[    0.382040] usbcore: registered new interface driver usbhid
[    0.382541] usbhid: USB HID core driver
[    0.384479] NET: Registered PF_INET6 protocol family
[    0.386463] Segment Routing with IPv6
[    0.386926] In-situ OAM (IOAM) with IPv6
[    0.387393] NET: Registered PF_PACKET protocol family
[    0.387934] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    0.389457] 8021q: 802.1Q VLAN Support v1.8
[    0.408036] vcc_sd: supplied by vcc_io
[    0.410865] rk_gmac-dwmac ff540000.ethernet: IRQ eth_wake_irq not found
[    0.411486] rk_gmac-dwmac ff540000.ethernet: IRQ eth_lpi not found
[    0.412169] rk_gmac-dwmac ff540000.ethernet: PTP uses main clock
[    0.412833] rk_gmac-dwmac ff540000.ethernet: clock input or output? (input).
[    0.413459] rk_gmac-dwmac ff540000.ethernet: TX delay(0x19).
[    0.413963] rk_gmac-dwmac ff540000.ethernet: RX delay(0x5).
[    0.414464] rk_gmac-dwmac ff540000.ethernet: integrated PHY? (no).
[    0.415039] rk_gmac-dwmac ff540000.ethernet: cannot get clock clk_mac_speed
[    0.415654] rk_gmac-dwmac ff540000.ethernet: clock input from PHY
[    0.421204] rk_gmac-dwmac ff540000.ethernet: init for RGMII
[    0.422070] rk_gmac-dwmac ff540000.ethernet: User ID: 0x10, Synopsys ID: 0x35
[    0.422779] rk_gmac-dwmac ff540000.ethernet: 	DWMAC1000
[    0.423246] rk_gmac-dwmac ff540000.ethernet: DMA HW capability register supported
[    0.423907] rk_gmac-dwmac ff540000.ethernet: RX Checksum Offload Engine supported
[    0.424568] rk_gmac-dwmac ff540000.ethernet: COE Type 2
[    0.425031] rk_gmac-dwmac ff540000.ethernet: TX Checksum insertion supported
[    0.425653] rk_gmac-dwmac ff540000.ethernet: Wake-Up On Lan supported
[    0.426223] rk_gmac-dwmac ff540000.ethernet: Normal descriptors
[    0.426780] rk_gmac-dwmac ff540000.ethernet: Ring mode enabled
[    0.427298] rk_gmac-dwmac ff540000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    0.603036] dwmmc_rockchip ff500000.mmc: IDMAC supports 32-bit address mode.
[    0.603096] dwmmc_rockchip ff500000.mmc: Using internal DMA controller.
[    0.604280] dwmmc_rockchip ff500000.mmc: Version ID is 270a
[    0.604825] dwmmc_rockchip ff500000.mmc: DW MMC controller at irq 34,32 bit host data width,256 deep fifo
[    0.612392] rockchip-pm-domain ff100000.syscon:power-controller: failed to get ack on domain 'hevc', val=0x88220
[    0.616739] mmc_host mmc0: Bus speed (slot 0) = 400000Hz (slot req 400000Hz, actual 400000HZ div = 0)
[    0.630960] Waiting for root device PARTUUID=5452574f-02...
[    0.666225] mmc_host mmc0: Bus speed (slot 0) = 50000000Hz (slot req 50000000Hz, actual 50000000HZ div = 0)
[    0.667221] mmc0: new high speed SDHC card at address 5048
[    0.668939] mmcblk0: mmc0:5048 SDCIT 7.49 GiB 
[    0.672534]  mmcblk0: p1 p2
[    0.694272] VFS: Mounted root (squashfs filesystem) readonly on device 179:2.
[    0.695872] Freeing unused kernel memory: 1664K
[    0.736855] Run /sbin/init as init process
[    0.976112] init: Console is alive
[    0.976867] init: - watchdog -
[    1.113793] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    1.130783] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    1.139720] init: - preinit -
[    1.528962] random: jshn: uninitialized urandom read (4 bytes read)
[    1.571230] random: jshn: uninitialized urandom read (4 bytes read)
[    1.603791] random: jshn: uninitialized urandom read (4 bytes read)
[    1.718465] dw-apb-uart ff130000.serial: forbid DMA for kernel console
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    2.756748] usb 4-1: new SuperSpeed USB device number 2 using xhci-hcd
[    3.788239] loop0: detected capacity change from 0 to 212992
[    3.846771] loop0: detected capacity change from 212992 to 206464
[    4.155251] F2FS-fs (loop0): Disable nat_bits due to incorrect cp_ver (6959071872808005461, 8080544892434132809)
[    4.162487] F2FS-fs (loop0): Mounted with checkpoint version = 61222f55
[    4.164339] mount_root: switching to f2fs overlay
[    4.231746] EXT4-fs (mmcblk0p1): mounted filesystem without journal. Opts: (null). Quota mode: none.
[    4.241401] urandom-seed: Seeding with /etc/urandom.seed
[    4.291024] procd: - early -
[    4.291452] procd: - watchdog -
[    4.855349] procd: - watchdog -
[    4.856458] procd: - ubus -
[    4.889750] random: ubusd: uninitialized urandom read (4 bytes read)
[    4.910419] random: ubusd: uninitialized urandom read (4 bytes read)
[    4.911632] random: ubusd: uninitialized urandom read (4 bytes read)
[    4.914848] procd: - init -
Please press Enter to activate this console.
[    5.184126] kmodloader: loading kernel modules from /etc/modules.d/*
[    5.200423] usbcore: registered new interface driver r8152
[    5.204156] usbcore: registered new interface driver cdc_ether
[    5.205911] usbcore: registered new interface driver cdc_ncm
[    5.236221] PPP generic driver version 2.4.2
[    5.237760] NET: Registered PF_PPPOX protocol family
[    5.243469] kmodloader: done loading kernel modules from /etc/modules.d/*
[    5.380870] urngd: v1.0.2 started.
[    5.406803] usb 4-1: reset SuperSpeed USB device number 2 using xhci-hcd
[    5.486537] r8152 4-1:1.0: load rtl8153b-2 v2 04/27/23 successfully
[    5.527808] r8152 4-1:1.0 eth1: v1.12.13
[    5.539054] random: crng init done
[    5.539401] random: 24 urandom warning(s) missed due to ratelimiting
[    8.591011] rk_gmac-dwmac ff540000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
[    8.658171] rk_gmac-dwmac ff540000.ethernet eth0: PHY [stmmac-0:00] driver [YT8531 Gigabit Ethernet] (irq=POLL)
[    8.666933] rk_gmac-dwmac ff540000.ethernet eth0: No Safety Features support found
[    8.667675] rk_gmac-dwmac ff540000.ethernet eth0: PTP not supported by HW
[    8.668924] rk_gmac-dwmac ff540000.ethernet eth0: configuring for phy/rgmii link mode
[    8.671210] br-lan: port 1(eth0) entered blocking state
[    8.671741] br-lan: port 1(eth0) entered disabled state
[    8.672577] device eth0 entered promiscuous mode
[    8.675257] br-lan: port 1(eth0) entered blocking state
[    8.675786] br-lan: port 1(eth0) entered forwarding state
[    8.692896] br-lan: port 2(eth1) entered blocking state
[    8.693430] br-lan: port 2(eth1) entered disabled state
[    8.694285] device eth1 entered promiscuous mode
[    8.695716] br-lan: port 2(eth1) entered blocking state
[    8.696251] br-lan: port 2(eth1) entered forwarding state
[    9.606985] br-lan: port 1(eth0) entered disabled state
[    9.607808] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[    9.609182] br-lan: port 2(eth1) entered disabled state
[   11.767495] rk_gmac-dwmac ff540000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   11.768412] br-lan: port 1(eth0) entered blocking state
[   11.768909] br-lan: port 1(eth0) entered forwarding state
[   12.115206] r8152 4-1:1.0 eth1: Promiscuous mode enabled
[   12.116025] r8152 4-1:1.0 eth1: carrier on
[   12.117190] br-lan: port 2(eth1) entered blocking state
[   12.117691] br-lan: port 2(eth1) entered forwarding state
```

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

Good day
I created a file in target/linux/rockchip/patches-5.15/ with custom content
--- a/arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts
@@ -38,3 +38,7 @@
  };
  };
  };
+
+&io_domains {
+ vccio4-supply = <&vcc_18>;
+};
But plowed the next mistake:
pplying /home/kuzya/openwrt/target/linux/rockchip/patches-5.15/rk3328-orangepi-r1-plus-lts.dts.patch using plaintext:
patching file arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts
patch unexpectedly ends in the middle of the line
patch: **** malformed patch at line 9:

Patch failed! Please fix /home/kuzya/openwrt/target/linux/rockchip/patches-5.15/rk3328-orangepi-r1-plus-lts.dts.patch!
make[4]: *** [Makefile:32: /home/kuzya/openwrt/build_dir/target-aarch64_generic_musl/linux-rockchip_armv8/linux-5.15.134/.prepared_6f603c91aef8fd38ebcd562374f29e0a] Error 1
make[4]: Leaving directory '/home/kuzya/openwrt/target/linux/rockchip'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/home/kuzya/openwrt/target/linux'
time: target/linux/compile#13.94#11.46#24.44
     ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:30: target/linux/compile] Error 1
make[2]: Leaving directory '/home/kuzya/openwrt'
make[1]: *** [target/Makefile:23: /home/kuzya/openwrt/staging_dir/target-aarch64_generic_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/home/kuzya/openwrt'
make: *** [/home/kuzya/openwrt/include/toplevel.mk:232: world]
Tell me what I'm doing wrong
Sorry for my English!!!!

--- a/target/linux/generic/backport-5.15/791-v6.6-11-net-phy-motorcomm-Add-pad-drive-strength-cfg-support.patch
+++ b/target/linux/generic/backport-5.15/791-v6.6-11-net-phy-motorcomm-Add-pad-drive-strength-cfg-support.patch
@@ -1,0 +1,170 @@
+From 7a561e9351ae7e3fb1f08584d40b49c1e55dde60 Mon Sep 17 00:00:00 2001
+From: Samin Guo <samin.guo@starfivetech.com>
+Date: Thu, 20 Jul 2023 19:15:09 +0800
+Subject: [PATCH] net: phy: motorcomm: Add pad drive strength cfg support
+
+The motorcomm phy (YT8531) supports the ability to adjust the drive
+strength of the rx_clk/rx_data, and the default strength may not be
+suitable for all boards. So add configurable options to better match
+the boards.(e.g. StarFive VisionFive 2)
+
+When we configure the drive strength, we need to read the current
+LDO voltage value to ensure that it is a legal value at that LDO
+voltage.
+
+Reviewed-by: Hal Feng <hal.feng@starfivetech.com>
+Signed-off-by: Samin Guo <samin.guo@starfivetech.com>
+Reviewed-by: Andrew Lunn <andrew@lunn.ch>
+Signed-off-by: David S. Miller <davem@davemloft.net>
+---
+ drivers/net/phy/motorcomm.c | 118 ++++++++++++++++++++++++++++++++++++
+ 1 file changed, 118 insertions(+)
+
+--- a/drivers/net/phy/motorcomm.c
++++ b/drivers/net/phy/motorcomm.c
+@@ -163,6 +163,10 @@
+ 
+ #define YT8521_CHIP_CONFIG_REG			0xA001
+ #define YT8521_CCR_SW_RST			BIT(15)
++#define YT8531_RGMII_LDO_VOL_MASK		GENMASK(5, 4)
++#define YT8531_LDO_VOL_3V3			0x0
++#define YT8531_LDO_VOL_1V8			0x2
++
+ /* 1b0 disable 1.9ns rxc clock delay  *default*
+  * 1b1 enable 1.9ns rxc clock delay
+  */
+@@ -236,6 +240,12 @@
+  */
+ #define YTPHY_WCR_TYPE_PULSE			BIT(0)
+ 
++#define YTPHY_PAD_DRIVE_STRENGTH_REG		0xA010
++#define YT8531_RGMII_RXC_DS_MASK		GENMASK(15, 13)
++#define YT8531_RGMII_RXD_DS_HI_MASK		BIT(12)		/* Bit 2 of rxd_ds */
++#define YT8531_RGMII_RXD_DS_LOW_MASK		GENMASK(5, 4)	/* Bit 1/0 of rxd_ds */
++#define YT8531_RGMII_RX_DS_DEFAULT		0x3
++
+ #define YTPHY_SYNCE_CFG_REG			0xA012
+ #define YT8521_SCR_SYNCE_ENABLE			BIT(5)
+ /* 1b0 output 25m clock
+@@ -835,6 +845,110 @@ static int ytphy_rgmii_clk_delay_config_
+ }
+ 
+ /**
++ * struct ytphy_ldo_vol_map - map a current value to a register value
++ * @vol: ldo voltage
++ * @ds:  value in the register
++ * @cur: value in device configuration
++ */
++struct ytphy_ldo_vol_map {
++	u32 vol;
++	u32 ds;
++	u32 cur;
++};
++
++static const struct ytphy_ldo_vol_map yt8531_ldo_vol[] = {
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 0, .cur = 1200},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 1, .cur = 2100},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 2, .cur = 2700},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 3, .cur = 2910},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 4, .cur = 3110},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 5, .cur = 3600},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 6, .cur = 3970},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 7, .cur = 4350},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 0, .cur = 3070},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 1, .cur = 4080},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 2, .cur = 4370},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 3, .cur = 4680},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 4, .cur = 5020},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 5, .cur = 5450},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 6, .cur = 5740},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 7, .cur = 6140},
++};
++
++static u32 yt8531_get_ldo_vol(struct phy_device *phydev)
++{
++	u32 val;
++
++	val = ytphy_read_ext_with_lock(phydev, YT8521_CHIP_CONFIG_REG);
++	val = FIELD_GET(YT8531_RGMII_LDO_VOL_MASK, val);
++
++	return val <= YT8531_LDO_VOL_1V8 ? val : YT8531_LDO_VOL_1V8;
++}
++
++static int yt8531_get_ds_map(struct phy_device *phydev, u32 cur)
++{
++	u32 vol;
++	int i;
++
++	vol = yt8531_get_ldo_vol(phydev);
++	for (i = 0; i < ARRAY_SIZE(yt8531_ldo_vol); i++) {
++		if (yt8531_ldo_vol[i].vol == vol && yt8531_ldo_vol[i].cur == cur)
++			return yt8531_ldo_vol[i].ds;
++	}
++
++	return -EINVAL;
++}
++
++static int yt8531_set_ds(struct phy_device *phydev)
++{
++	struct device_node *node = phydev->mdio.dev.of_node;
++	u32 ds_field_low, ds_field_hi, val;
++	int ret, ds;
++
++	/* set rgmii rx clk driver strength */
++	if (!of_property_read_u32(node, "motorcomm,rx-clk-drv-microamp", &val)) {
++		ds = yt8531_get_ds_map(phydev, val);
++		if (ds < 0)
++			return dev_err_probe(&phydev->mdio.dev, ds,
++					     "No matching current value was found.\n");
++	} else {
++		ds = YT8531_RGMII_RX_DS_DEFAULT;
++	}
++
++	ret = ytphy_modify_ext_with_lock(phydev,
++					 YTPHY_PAD_DRIVE_STRENGTH_REG,
++					 YT8531_RGMII_RXC_DS_MASK,
++					 FIELD_PREP(YT8531_RGMII_RXC_DS_MASK, ds));
++	if (ret < 0)
++		return ret;
++
++	/* set rgmii rx data driver strength */
++	if (!of_property_read_u32(node, "motorcomm,rx-data-drv-microamp", &val)) {
++		ds = yt8531_get_ds_map(phydev, val);
++		if (ds < 0)
++			return dev_err_probe(&phydev->mdio.dev, ds,
++					     "No matching current value was found.\n");
++	} else {
++		ds = YT8531_RGMII_RX_DS_DEFAULT;
++	}
++
++	ds_field_hi = FIELD_GET(BIT(2), ds);
++	ds_field_hi = FIELD_PREP(YT8531_RGMII_RXD_DS_HI_MASK, ds_field_hi);
++
++	ds_field_low = FIELD_GET(GENMASK(1, 0), ds);
++	ds_field_low = FIELD_PREP(YT8531_RGMII_RXD_DS_LOW_MASK, ds_field_low);
++
++	ret = ytphy_modify_ext_with_lock(phydev,
++					 YTPHY_PAD_DRIVE_STRENGTH_REG,
++					 YT8531_RGMII_RXD_DS_LOW_MASK | YT8531_RGMII_RXD_DS_HI_MASK,
++					 ds_field_low | ds_field_hi);
++	if (ret < 0)
++		return ret;
++
++	return 0;
++}
++
++/**
+  * yt8521_probe() - read chip config then set suitable polling_mode
+  * @phydev: a pointer to a &struct phy_device
+  *
+@@ -1518,6 +1632,10 @@ static int yt8531_config_init(struct phy
+ 			return ret;
+ 	}
+ 
++	ret = yt8531_set_ds(phydev);
++	if (ret < 0)
++		return ret;
++
+ 	return 0;
+ }
+ 

--- a/target/linux/generic/backport-6.1/791-v6.6-11-net-phy-motorcomm-Add-pad-drive-strength-cfg-support.patch
+++ b/target/linux/generic/backport-6.1/791-v6.6-11-net-phy-motorcomm-Add-pad-drive-strength-cfg-support.patch
@@ -1,0 +1,170 @@
+From 7a561e9351ae7e3fb1f08584d40b49c1e55dde60 Mon Sep 17 00:00:00 2001
+From: Samin Guo <samin.guo@starfivetech.com>
+Date: Thu, 20 Jul 2023 19:15:09 +0800
+Subject: [PATCH] net: phy: motorcomm: Add pad drive strength cfg support
+
+The motorcomm phy (YT8531) supports the ability to adjust the drive
+strength of the rx_clk/rx_data, and the default strength may not be
+suitable for all boards. So add configurable options to better match
+the boards.(e.g. StarFive VisionFive 2)
+
+When we configure the drive strength, we need to read the current
+LDO voltage value to ensure that it is a legal value at that LDO
+voltage.
+
+Reviewed-by: Hal Feng <hal.feng@starfivetech.com>
+Signed-off-by: Samin Guo <samin.guo@starfivetech.com>
+Reviewed-by: Andrew Lunn <andrew@lunn.ch>
+Signed-off-by: David S. Miller <davem@davemloft.net>
+---
+ drivers/net/phy/motorcomm.c | 118 ++++++++++++++++++++++++++++++++++++
+ 1 file changed, 118 insertions(+)
+
+--- a/drivers/net/phy/motorcomm.c
++++ b/drivers/net/phy/motorcomm.c
+@@ -163,6 +163,10 @@
+ 
+ #define YT8521_CHIP_CONFIG_REG			0xA001
+ #define YT8521_CCR_SW_RST			BIT(15)
++#define YT8531_RGMII_LDO_VOL_MASK		GENMASK(5, 4)
++#define YT8531_LDO_VOL_3V3			0x0
++#define YT8531_LDO_VOL_1V8			0x2
++
+ /* 1b0 disable 1.9ns rxc clock delay  *default*
+  * 1b1 enable 1.9ns rxc clock delay
+  */
+@@ -236,6 +240,12 @@
+  */
+ #define YTPHY_WCR_TYPE_PULSE			BIT(0)
+ 
++#define YTPHY_PAD_DRIVE_STRENGTH_REG		0xA010
++#define YT8531_RGMII_RXC_DS_MASK		GENMASK(15, 13)
++#define YT8531_RGMII_RXD_DS_HI_MASK		BIT(12)		/* Bit 2 of rxd_ds */
++#define YT8531_RGMII_RXD_DS_LOW_MASK		GENMASK(5, 4)	/* Bit 1/0 of rxd_ds */
++#define YT8531_RGMII_RX_DS_DEFAULT		0x3
++
+ #define YTPHY_SYNCE_CFG_REG			0xA012
+ #define YT8521_SCR_SYNCE_ENABLE			BIT(5)
+ /* 1b0 output 25m clock
+@@ -835,6 +845,110 @@ static int ytphy_rgmii_clk_delay_config_
+ }
+ 
+ /**
++ * struct ytphy_ldo_vol_map - map a current value to a register value
++ * @vol: ldo voltage
++ * @ds:  value in the register
++ * @cur: value in device configuration
++ */
++struct ytphy_ldo_vol_map {
++	u32 vol;
++	u32 ds;
++	u32 cur;
++};
++
++static const struct ytphy_ldo_vol_map yt8531_ldo_vol[] = {
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 0, .cur = 1200},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 1, .cur = 2100},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 2, .cur = 2700},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 3, .cur = 2910},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 4, .cur = 3110},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 5, .cur = 3600},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 6, .cur = 3970},
++	{.vol = YT8531_LDO_VOL_1V8, .ds = 7, .cur = 4350},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 0, .cur = 3070},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 1, .cur = 4080},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 2, .cur = 4370},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 3, .cur = 4680},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 4, .cur = 5020},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 5, .cur = 5450},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 6, .cur = 5740},
++	{.vol = YT8531_LDO_VOL_3V3, .ds = 7, .cur = 6140},
++};
++
++static u32 yt8531_get_ldo_vol(struct phy_device *phydev)
++{
++	u32 val;
++
++	val = ytphy_read_ext_with_lock(phydev, YT8521_CHIP_CONFIG_REG);
++	val = FIELD_GET(YT8531_RGMII_LDO_VOL_MASK, val);
++
++	return val <= YT8531_LDO_VOL_1V8 ? val : YT8531_LDO_VOL_1V8;
++}
++
++static int yt8531_get_ds_map(struct phy_device *phydev, u32 cur)
++{
++	u32 vol;
++	int i;
++
++	vol = yt8531_get_ldo_vol(phydev);
++	for (i = 0; i < ARRAY_SIZE(yt8531_ldo_vol); i++) {
++		if (yt8531_ldo_vol[i].vol == vol && yt8531_ldo_vol[i].cur == cur)
++			return yt8531_ldo_vol[i].ds;
++	}
++
++	return -EINVAL;
++}
++
++static int yt8531_set_ds(struct phy_device *phydev)
++{
++	struct device_node *node = phydev->mdio.dev.of_node;
++	u32 ds_field_low, ds_field_hi, val;
++	int ret, ds;
++
++	/* set rgmii rx clk driver strength */
++	if (!of_property_read_u32(node, "motorcomm,rx-clk-drv-microamp", &val)) {
++		ds = yt8531_get_ds_map(phydev, val);
++		if (ds < 0)
++			return dev_err_probe(&phydev->mdio.dev, ds,
++					     "No matching current value was found.\n");
++	} else {
++		ds = YT8531_RGMII_RX_DS_DEFAULT;
++	}
++
++	ret = ytphy_modify_ext_with_lock(phydev,
++					 YTPHY_PAD_DRIVE_STRENGTH_REG,
++					 YT8531_RGMII_RXC_DS_MASK,
++					 FIELD_PREP(YT8531_RGMII_RXC_DS_MASK, ds));
++	if (ret < 0)
++		return ret;
++
++	/* set rgmii rx data driver strength */
++	if (!of_property_read_u32(node, "motorcomm,rx-data-drv-microamp", &val)) {
++		ds = yt8531_get_ds_map(phydev, val);
++		if (ds < 0)
++			return dev_err_probe(&phydev->mdio.dev, ds,
++					     "No matching current value was found.\n");
++	} else {
++		ds = YT8531_RGMII_RX_DS_DEFAULT;
++	}
++
++	ds_field_hi = FIELD_GET(BIT(2), ds);
++	ds_field_hi = FIELD_PREP(YT8531_RGMII_RXD_DS_HI_MASK, ds_field_hi);
++
++	ds_field_low = FIELD_GET(GENMASK(1, 0), ds);
++	ds_field_low = FIELD_PREP(YT8531_RGMII_RXD_DS_LOW_MASK, ds_field_low);
++
++	ret = ytphy_modify_ext_with_lock(phydev,
++					 YTPHY_PAD_DRIVE_STRENGTH_REG,
++					 YT8531_RGMII_RXD_DS_LOW_MASK | YT8531_RGMII_RXD_DS_HI_MASK,
++					 ds_field_low | ds_field_hi);
++	if (ret < 0)
++		return ret;
++
++	return 0;
++}
++
++/**
+  * yt8521_probe() - read chip config then set suitable polling_mode
+  * @phydev: a pointer to a &struct phy_device
+  *
+@@ -1518,6 +1632,10 @@ static int yt8531_config_init(struct phy
+ 			return ret;
+ 	}
+ 
++	ret = yt8531_set_ds(phydev);
++	if (ret < 0)
++		return ret;
++
+ 	return 0;
+ }
+ 

--- a/target/linux/rockchip/patches-6.1/015-v6.8-arm64-dts-rockchip-configure-eth-pad-driver-strength-for-.patch
+++ b/target/linux/rockchip/patches-6.1/015-v6.8-arm64-dts-rockchip-configure-eth-pad-driver-strength-for-.patch
@@ -1,0 +1,33 @@
+From fc5a80a432607d05e85bba37971712405f75c546 Mon Sep 17 00:00:00 2001
+From: Tianling Shen <cnsztl@gmail.com>
+Date: Sat, 16 Dec 2023 12:07:23 +0800
+Subject: [PATCH] arm64: dts: rockchip: configure eth pad driver strength
+ for orangepi r1 plus lts
+
+The default strength is not enough to provide stable connection
+under 3.3v LDO voltage.
+
+Fixes: 387b3bbac5ea ("arm64: dts: rockchip: Add Xunlong OrangePi R1 Plus LTS")
+Cc: stable@vger.kernel.org # 6.6+
+Signed-off-by: Tianling Shen <cnsztl@gmail.com>
+Link: https://lore.kernel.org/r/20231216040723.17864-1-cnsztl@gmail.com
+Signed-off-by: Heiko Stuebner <heiko@sntech.de>
+---
+ arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts | 4 +++-
+ 1 file changed, 3 insertions(+), 1 deletion(-)
+
+--- a/arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts
++++ b/arch/arm64/boot/dts/rockchip/rk3328-orangepi-r1-plus-lts.dts
+@@ -26,9 +26,11 @@
+ 			compatible = "ethernet-phy-ieee802.3-c22";
+ 			reg = <0>;
+ 
++			motorcomm,auto-sleep-disabled;
+ 			motorcomm,clk-out-frequency-hz = <125000000>;
+ 			motorcomm,keep-pll-enabled;
+-			motorcomm,auto-sleep-disabled;
++			motorcomm,rx-clk-drv-microamp = <5020>;
++			motorcomm,rx-data-drv-microamp = <5020>;
+ 
+ 			pinctrl-0 = <&eth_phy_reset_pin>;
+ 			pinctrl-names = "default";

